File: 1520A/main.cpp
#include <iostream>
#include <vector>
#include <bits/stdc++.h>
using namespace std;
int main() {
    int t;
    cin >> t;
    while(t--){
        int n;
        string s;
        cin >> n >> s;
        set<char> counts;
        char curr = s[0];
        bool res = false;
        for(auto& a : s){
            if (a!=curr and counts.count(a)){
                res = true;
                break;
            }
            
            counts.insert(a);
            curr = a;
        }

        if (res) {
            cout << "NO" << endl;
        } else {
            cout << "YES" << endl;
        }
    }
    
    // for(auto& i:v){
    //     cout << i << endl;
    // }
    
    return 0;
}

File: 1520A/main.py
l=[]
t = int(input())
for i in range(t):
    n = int(input())
    s = input()
    curr = s[0]
    c = set()
    flag = False
    for i in s:
        if i!=curr and i in c:
            flag = True
            break
        c.add(i)
        curr = i
    if flag:
        print("NO")
    else:
        print("YES")
# for i in l:
#     print(i)